### PR TITLE
Update neighbour_letter model to permit resending

### PR DESCRIPTION
**Pull request by benjamineskola**
### Description of change

- Make neighbour->neighbour_letter a one-to-many relationship
- Add resend reason to neighbour letter
- Add last_letter_sent_at to neighbour, and update it from LetterSendingService

### Story Link

https://trello.com/c/csQTNIyO/1829-repeat-previous-publicity-tasks-at-any-point-of-the-planning-process

### CODE REVIEW

**Comment by benjamineskola** [comment]
[Repeat previous Publicity tasks at any point of the planning process](https://trello.com/c/csQTNIyO/1829-repeat-previous-publicity-tasks-at-any-point-of-the-planning-process)

File: app/models/neighbour_letter.rb [hunk 1]
```diff
@@ -3,6 +3,9 @@
 class NeighbourLetter < ApplicationRecord
   belongs_to :neighbour
 
+  validates :resend_reason, absence: true, on: :create, if: -> { neighbour.last_letter_sent_at.nil? }
+  validates :resend_reason, presence: true, on: :create, if: -> { neighbour.last_letter_sent_at.present? }
+
```
Comment (pixeltrix) [inline app/models/neighbour_letter.rb:8]: Could do these as:

``` ruby
validates :resend_reason, absence: true, on: :create, unless: :resend?
validates :resend_reason, presence: true, on: :create, if: :resend?

private

  def resend?
    neighbour.last_letter_sent_at?
  end
```

Should the validations be `on: :create` or do they need to be on `update` as well?

**Review by pixeltrix — APPROVED** [review]
Just a couple of suggestions

File: app/models/neighbour_letter.rb (hunk 1 above)
Comment (benjamineskola) [reply app/models/neighbour_letter.rb:8]: I'm not sure if we're ever likely to modify the letter object after creation but it can't hurt to make certain.

Comment (pixeltrix) [reply app/models/neighbour_letter.rb:8]: you can just remove `on`

**Pull request merged**